### PR TITLE
Release network resources on container stop

### LINKED ISSUE #992: docker rm and stop do not release network
Currently `docker rm` and `docker stop` do not do a network disconnect and unbind respectively. This results in `network rm` not being able to delete a user defined network.

### PR DESCRIPTION
**Pull request by hmahmood**
Fixes #992

### CODE REVIEW

File: lib/portlayer/network/context.go [hunk 1]
```diff
@@ -391,49 +395,60 @@ func (c *Context) Scopes(idName *string) ([]*Scope, error) {
...
+		s, ok := c.scopes[ne.Network.Name]
+		if !ok {
+			return nil, &ResourceNotFoundError{}
 		}
 
-		for _, s := range bound {
-			s.unbindContainer(con)
-		}
-	}()
+		defer func() {
```
Comment (caglar10ur) [inline lib/portlayer/network/context.go]: Why do we have this function?

Comment (hmahmood) [inline lib/portlayer/network/context.go]: To roll back the `addContainer` calls (below) that have happened to date in case of error.

Comment (caglar10ur) [inline lib/portlayer/network/context.go]: Ahh I thought 
```
defer func() {
		if err == nil {
 				return
 			}
}
```

is the whole function...

**Comment by caglar10ur** [comment]
LGTM :sheep: :it:

**Comment by matthewavery** [comment]
LGTM

**Comment by hmahmood** [comment]
lgtm

**Pull request merged**